h2o-core/src/main/java/water/udf/CDistributionFunc.java

```diff
@@ -35,6 +36,16 @@
      */
     double gradient(double y, double f);
 
+    /**
+     * (Negative half) Gradient of deviance function at predicted value f, for actual response y.
+     * Important for customization of a loss function.
+     * @param y (actual) response
+     * @param f (predicted) response in link space (including offset)
+     * @param l (class label) label of a class (converted lexicographically from original labels to 0-number of class - 1)  
+     * @return gradient
+     */
+    double lgradient(double y, double f, int l);
```

Comment:
I decided to add a new method rather than add a new parameter to the gradient method. It is compatible with the old code, and it allows using a class label to customize gradient calculation now. 

WDYT @michalkurka?

Reply:
Finally, I removed `lgradient` method and override gradient method.